File: RaftSwift/Sources/RaftCore/Utils/Types/LeaderState.swift
/// Leader only volatile state tracking next log entry to send to each follower
struct LeaderState {
    /// For each follower, index of the next log entry to send to that follower
    var nextIndex: [Int: Int] = [:]

    /// For each follower, index of highest log entry known to be replicated on that follower
    var matchIndex: [Int: Int] = [:]
}

File: RaftKotlin/raft-node/src/main/kotlin/raft/utils/types/LeaderState.kt
package raft.utils.types

/**
 * Leader only volatile state tracking next log entry to send to each follower
 */
data class LeaderState(
    /**
     * For each follower, index of the next log entry to send to that follower
     */
    var nextIndex: MutableMap<Int, Int> = mutableMapOf(),
    /**
     * For each follower, index of highest log entry known to be replicated on that follower
     */
    var matchIndex: MutableMap<Int, Int> = mutableMapOf()
)
